src/Collection/CollectionTrait.php

```diff
@@ -658,4 +658,22 @@ public function _unwrap()
     {
         return $this->unwrap();
     }
+
+    /**
+     * {@inheritDoc}
+     *
+     * @return \Cake\Collection
+     */
+    public function transpose()
+    {
+        $length = count($this->first());
+        $result = [];
+        foreach ($this->toArray() as $column => $row) {
+            if (count($row) != $length) {
+                throw new \LogicException('Child arrays do not have even length');
+            }
+            $result[] = array_column($this->toArray(), $column);
```

Comment:
You're calling `toArray()` multiple times. You should probably keep a local variable with the original value. `toArray()` is not a cheap method to run.